docs/additional/interesting_aspects_de.md

```diff
@@ -0,0 +1,248 @@
+Der Ordner wird mit derselben Struktur wie das Container-Dateisystem aufgebaut.
+
+Beispiel:
+- `resources/`
+  - `create-sa.sh` (optional)
+  - `remove-sa.sh` (optional)
+  - `post-upgrade.sh` (optional)
+  - `pre-upgrade.sh` (optional)
+  - `upgrade-notification.sh` (optional)
+  - `startup.sh` (empfohlen)
+
+Das Beispiel zeigt den Inhalt eines resource-Ordners eines Dogus.
+
+Der komplette Ordner wird im Dockerfile in das Root-Verzeichnis des Dogu-Containers kopiert.  
+Damit würde eine vorhandene Konfiguration der App im Pfad `/etc/your_app/config.json` überschrieben werden.  
+Durch diesen Ansatz ist es möglich alle Dateien des Containers anzupassen.
+
+### Bash-Skripte
+
+Das Anlegen von Bash-Skripten in einem Dogu ist notwendig, um verschiedene Abläufe zu steuern.  
+Darunter fällt das Starten und Upgraden des Dogus, sowie das Erstellen eines Service Accounts.
+
+Generell sollten alle Skripte folgende Konventionen erfüllen:
+
+#### Bash-Strict-Mode
+
+Bei Bash-Skripten soll immer der [Bash-Strict-Mode][strict-mode] verwendet werden.  
+Dieser verhindert bei einem Fehlverhalten die weitere Ausführung des Skripts.
+Generell ist es sinnvoll, alle potenziellen Fehler im Startskript abzufangen und mit einer eigenen Fehlermethode zu berichten.  
+Diese könnte folgendermaßen aussehen:
+
+```shell
+# This function prints an error to the console and waits 5 minutes before exiting the process.
+function exitOnErrorWithMessage() {
+  message=${1}
+  echo -e "ERROR: ${message}. Exiting in 300 seconds."
+  sleep 300
+  exit 1
+}
+```
+
+Solch eine Methode macht das komplette Skript lesbarer und bremst den Neustart-Loop des Dogus sowie die Masse der generierten Logs.
+
+[strict-mode]: http://redsymbol.net/articles/unofficial-bash-strict-mode/
+
+#### Line-Endings
+
+Für alle Skripte sollten jederzeit Unix-basierte Line-Endings benutzt werden.
+
+#### Doguctl verwenden
+
+Doguctl ist ein Kommandozeilentool, um die Konfiguration eines Dogus zu vereinfachen.  
+Informationen über die Benutzung finden sich in ["Die Nutzung von doguctl"][doguctl-usage].
+
+Wo anwendbar, soll immer Doguctl in Bash-Skripten wie der `startup.sh` verwendet werden.  
+Für die sinnvolle Benutzung kann das [Startup-Skript vom Nexus][doguctl-example] als Beispiel genommen werden.
+
+[doguctl-usage]: ../important/relevant_functionalities_de.md#die-nutzung-von-doguctl
+[doguctl-example]: https://github.com/cloudogu/nexus/blob/4d2de3733eca684df1363179c527363a4d31526e/resources/startup.sh
+
+### Dogu-Startskript: startup.sh
+
+Das [Dogu-Start-Skript][startup-sh] ist der Einstiegspunkt eines Dogu-Containers und verwaltet alle Schritte, die für einen reibungslosen Start des Dogus notwendig sind.
+Konventionell wird das Startup-Skript mit dem Namen `startup.sh` in dem `resources`-Ordner des Dogus angelegt.
+
+[startup-sh]: ../important/relevant_functionalities_de.md#aufbau-und-best-practices-von-startupsh
+
+## Schnelle Feedback-Zyklen durch Tool-Entwicklung ohne CES
```

Comment:
- Title length and focus
- abbreviation of our product CES vs Cloudogu EcoSystem

This subchapter title is quite long while focussing on one paragraph about developing without EcoSystem. I feel the title would still make sense when shortened to:

```suggestion
## Schnelle Feedback-Zyklen
```